### PR TITLE
Add retries for getting yamls from Github

### LINKED ISSUE #13: Add retries for getting yamls from Github
There are several places where we pull manifests from Github (i.e. [here](https://github.com/knative-sandbox/kn-plugin-quickstart/blob/main/pkg/install/install.go#L73), [here](https://github.com/knative-sandbox/kn-plugin-quickstart/blob/main/pkg/install/install.go#L84), and [here](https://github.com/knative-sandbox/kn-plugin-quickstart/blob/main/pkg/install/install.go#L30), though there may be others as well). 

Currently, the plugin only tries getting the manifest once, but to make the plugin more resilient we should add retries to mitigate against network issues that may cause a single try to fail.

### PR DESCRIPTION
**Pull request by psschwei**
# Changes

- Adds retries when applying yamls from Github via kubectl

/kind enhancement

Fixes #13

### CODE REVIEW

File: pkg/install/install.go [hunk 1]
```diff
@@ -29,17 +30,19 @@ func Kourier() error {
 	fmt.Println("Starting Networking layer install...")
 
 	kourier := exec.Command("kubectl", "apply", "-f", "https://github.com/knative-sandbox/net-kourier/releases/download/v"+kourierVersion+"/kourier.yaml")
-	if err := kourier.Run(); err != nil {
-		return fmt.Errorf("apply: %w", err)
+	for i := 0; i <= 2; {
+		kourier.Run()
+		i++
```
Comment (julz) [inline pkg/install/install.go:35]: why not do this in the for loop?

**Review by csantanapr — APPROVED** [review]
❤️ thanks for adding this retries
/lgtm
/approve

File: pkg/install/install.go [hunk 2]
```diff
@@ -29,17 +30,19 @@ func Kourier() error {
 	fmt.Println("Starting Networking layer install...")
 
 	kourier := exec.Command("kubectl", "apply", "-f", "https://github.com/knative-sandbox/net-kourier/releases/download/v"+kourierVersion+"/kourier.yaml")
-	if err := kourier.Run(); err != nil {
-		return fmt.Errorf("apply: %w", err)
+	for i := 0; i <= 2; {
```
Comment (julz) [inline pkg/install/install.go:33]: I don't understand this loop, it doesnt check any errors so it always runs the download three times, and always takes 15 seconds?

Comment (psschwei) [reply pkg/install/install.go:33]: Should probably have been something like this:
```go
        for i := 0; i <= 2; i++ {
		if err := kourier.Run(); err != nil {
			if i >=2 {
				return fmt.Errorf("apply: %w", err)
			}
			time.Sleep(5 * time.Second)
		} else {
			break
		}
	}
```

Comment (julz) [reply pkg/install/install.go:33]: you already have apimachinery in go.sum, so you might be better off using one of the [wait.Poll](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait) methods from there.

Comment (psschwei) [reply pkg/install/install.go:33]: I'm probably missing something obvious, but what would be the ConditionFunc in that scenario?

Comment (julz) [reply pkg/install/install.go:33]: Something like:

~~~
if err := wait.PollImmediate(1 * time.Second, 5 * time.Second, func() (bool, error) {
   return kourier.Run() == nil, nil
}; err != nil {
   return fmt.Errorf("wait: %w", lastErr)
}
~~~

**Pull request merged**